### PR TITLE
Adding support for include in map and support some proxy_cache options

### PR DESCRIPTION
**Pull request by ceonizm**
Hello, 
I needed to use the include directive in a map block to handle some redirections that were dynamically generated by an other app. I noticed this feature was missing from your module so I've added it.

I've also added the support for proxy_cache_bypass and proxy_cache_lock 

Hopes it helps.

Best Regards
François Boukhalfa

### CODE REVIEW

File: templates/conf.d/map.erb [hunk 1]
```diff
@@ -6,6 +6,11 @@ map <%= @string %> $<%= @name %> {
 <% if @default -%>
   default <%= @default %>;
 <% end -%>
+<% if @includes.is_a?(Array) -%>
```
Comment (alexjfisher) [inline templates/conf.d/map.erb]: Can it ever not be an array?

File: templates/server/locations/proxy.erb [hunk 2]
```diff
@@ -46,4 +46,12 @@
 <% if @proxy_cache_key -%>
     proxy_cache_key         <%= @proxy_cache_key %>;
 <% end -%>
+<% if @proxy_cache_bypass && Array(@proxy_cache_bypass).size > 0 -%>
```
Comment (alexjfisher) [inline templates/server/locations/proxy.erb]: Is the size check needed?

**Comment by alexjfisher** [comment]
@ceonizm thanks! Would it be possible to split this PR up by feature you’re adding? Eg support for access log in a location seems unrelated to the proxy_cache changes.

File: templates/server/locations/proxy.erb (hunk 2 above)
Comment (ceonizm) [reply templates/server/locations/proxy.erb]: Initially when I've added theses features I've started from the 0.5.0 version which was published on forge and followed the same coding style, that's why there is this check I believe: 
I totally agree with you it seems useless now.

**Comment by ceonizm** [comment]
I close this PR and open some new one from different branches in order to get all features splitted.
- #1149 
- #1150 
- #1151

**Comment by alexjfisher** [comment]
@ceonizm Thanks.  This will make it easier to review and get your additions merged.

**Pull request closed without merging**